resources/configure_dependencies.py

```diff
@@ -1,7 +1,14 @@
 """
 This is a utility script for the download and installation of OpenSn
 dependencies.
+
+NOTES for clang compiler:
+- CFLAGS, CXXFLAGS and FFLAGS passed into PETSc won't work. Unset them.
```

Comment:
In the PETSc build we explicitly set the CFLAGS, CXXFLAGS, and FFLAGS without a compiler check. Could you explain this a little more for me?

Reply:
OpenMP flag used by `gcc` is `-fopenmp`. `clang` uses `-fopenmp=libomp`. But, I think I was not super clear with this comment. When I used `clang` from `conda`, PETSc just would not configure. This is probably caused by conda exporting its own compiler flags into the environment, so the notes should be for "clang in conda environment".

As far as handling the PIC and OpenMP flags, PETSc has dedicated options for those (they are `--with-pic` and `--with-openmp`). I think this script should really use those instead. I think I want to add a patch for this change to this PR...

Reply:
Roger that. Sounds good to me

Reply:
I updated the script to use `--with-pic` (well, it was already in the flags), and added `--with-openmp`. I dropped `CFLAGS`, `CXXFLAGS` and `FCFLAGS`, so now they are set by PETSc (and work for all PETSc supported compilers). 

This opened another interesting problem: clang does not support `-march=native` which is specified in the OPTFLAGS, so I added some more notes on that.

But, it really seems that all this knowledge should be documented in the doco on pages dedicated to builds on individual platforms. I don't think we want to waste our time on maintaining this script and making sure it works everywhere...

Reply:
while we wait for the doco, I think we should go through the 4 install instructions we have: MacOS/Linux Easy/Standard.

I am in favor of supportting only the Standard versions for now. These do NOT rely on the configure-deps script at all. However, int he standard instructions, we can add a note that a canned version is possible through the scripts, without warranty. 

Is this ok for now? Can we do better? FYI, OpenMC does this: https://docs.openmc.org/en/stable/quickinstall.html

Reply:
I suggest to move this discussion into its own issue - it seems we need to make somewhat bigger changes.